src/main/java/com/codeofus/rent_a_park/services/PersonService.java

```diff
@@ -22,17 +25,22 @@ public class PersonService {
 
     PersonRepository personRepository;
 
-    ParkingMapper mapper;
-
-    public List<PersonDto> getAll() {
-        return personRepository.findAll().stream().map(mapper::personToDto).toList();
-    }
-
     @Transactional
     public Person addNewPerson(Person person) {
         return personRepository.save(person);
     }
 
+    public List<Person> getAllPersons(Integer pageNo, Integer pageSize, String sortBy) {
+        Pageable pageable = PageRequest.of(pageNo, pageSize, Sort.by(sortBy));
+        Page<Person> pagedResult = personRepository.findAll(pageable);
+
+        if (pagedResult.hasContent()) {
+            return pagedResult.getContent();
+        } else {
+            return new ArrayList<>();
```

Comment:
Whenever returning an empty list (that you have no intentions in filling) return "List.of()" because it's immutable and every time you call the "List.of()" you actually get the same instance so it's more performant/memory friendly